src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,17 +57,17 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request: Publish => {
```

Comment:
when you use a pattern like Foobar(_) then it will match using Foobar's unapply, which is typesafe and all. when matching with a pattern f: Foobar, it's a whole different matching mechanism, that is less typsafe. it doesn't really matter here, but the rule of thumb is that we don't match on type.